stan/math/rev/fun/unit_vector_constrain.hpp

```diff
@@ -47,51 +47,54 @@ class unit_vector_elt_vari : public vari {
  * Return the unit length vector corresponding to the free vector y.
  * See https://en.wikipedia.org/wiki/N-sphere#Generating_random_points
  *
- * @tparam R number of rows, can be Eigen::Dynamic
- * @tparam C number of columns, can be Eigen::Dynamic
+ * @tparam EigMat type inheriting from `EigenBase` that has a `var`
+ *  scalar type.
  * @param y vector of K unrestricted variables
  * @return Unit length vector of dimension K
  **/
-template <int R, int C>
-Eigen::Matrix<var, R, C> unit_vector_constrain(
-    const Eigen::Matrix<var, R, C>& y) {
+template <typename EigMat, require_eigen_vt<is_var, EigMat>* = nullptr>
+auto unit_vector_constrain(EigMat&& y) {
+  using ref_inner = const typename std::decay_t<EigMat>::PlainObject;
   check_vector("unit_vector", "y", y);
   check_nonzero_size("unit_vector", "y", y);
-  vector_d y_d = y.val();
+  const Eigen::Ref<ref_inner, Eigen::Aligned16, Eigen::Stride<0, 0>>& y_mat = y;
```

Comment:
How can you be sure about the alignment? Also why do you need to explicitely set stride? Are not 0,0 default values anyway?

Why not just use `ref_type_t` or `to_ref`?

Reply:
Yes i wrote this before we had `to_ref` (which is soooo much easier)